### PR TITLE
In call hierarchy look through variable accessors and show surrounding function call

### PR DESCRIPTION
**Pull request by ahoppen**
Fix two bugs in call hierarchy:
1. In the following, we used to show `myVar` as the caller of `foo`. That was a dead end and didn’t allow any further exploration
```swift
func foo() {}
func testFunc(x: String) {
  let myVar = foo()
}
```
2. Handle variables and their accessors in call hierarchy

rdar://123526105

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Sources/SourceKitLSP/SourceKitServer.swift [hunk 1]
```diff
@@ -2031,8 +2032,9 @@ extension SourceKitServer {
     else {
       return []
     }
-    let occurs = index.occurrences(relatedToUSR: data.usr, roles: .calledBy)
-    let calls = occurs.compactMap { occurrence -> CallHierarchyOutgoingCall? in
+    let callableUsrs = [data.usr] + index.occurrences(relatedToUSR: data.usr, roles: .accessorOf).map(\.symbol.usr)
+    let callOccurrances = callableUsrs.flatMap { index.occurrences(relatedToUSR: $0, roles: .calledBy) }
```
Comment (bnbarham) [inline Sources/SourceKitLSP/SourceKitServer.swift:2036]:
```suggestion
    let callOccurrences = callableUsrs.flatMap { index.occurrences(relatedToUSR: $0, roles: .calledBy) }
```

File: Sources/SourceKitLSP/SourceKitServer.swift [hunk 2]
```diff
@@ -2000,27 +2000,28 @@ extension SourceKitServer {
...
       return []
     }
-    let occurs = index.occurrences(ofUSR: data.usr, roles: .calledBy)
-    let calls = occurs.compactMap { occurrence -> CallHierarchyIncomingCall? in
-      guard let location = indexToLSPLocation(occurrence.location),
-        let related = occurrence.relations.first
-      else {
-        return nil
+    let callableUsrs = [data.usr] + index.occurrences(relatedToUSR: data.usr, roles: .accessorOf).map(\.symbol.usr)
+    let callOccurrences = callableUsrs.flatMap { index.occurrences(ofUSR: $0, roles: .calledBy) }
```
Comment (bnbarham) [inline Sources/SourceKitLSP/SourceKitServer.swift:2004]: Does this ignore eg. `let foo = someFunc`? I know it's not strictly correct, but IMO it's still useful to include non-calls in call hierarchy. Open to opinions though.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**